mavutil.py

```diff
@@ -2004,7 +1983,60 @@ def mode_string_v09(msg):
     9: 'SURFACE',
     16: 'POSHOLD',
     19: 'MANUAL',
-    }
+}
+
+AP_MAV_TYPE_MODE_MAP_DEFAULT = {
+    # copter
+    mavlink.MAV_TYPE_HELICOPTER:  mode_mapping_acm,
+    mavlink.MAV_TYPE_TRICOPTER:   mode_mapping_acm,
+    mavlink.MAV_TYPE_QUADROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_HEXAROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_OCTOROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_DECAROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_DODECAROTOR: mode_mapping_acm,
+    mavlink.MAV_TYPE_COAXIAL:     mode_mapping_acm,
+    # plane
+    mavlink.MAV_TYPE_FIXED_WING: mode_mapping_apm,
+    # rover
+    mavlink.MAV_TYPE_GROUND_ROVER: mode_mapping_rover,
+    # boat
+    mavlink.MAV_TYPE_SURFACE_BOAT: mode_mapping_rover, # for the time being
+    # tracker
+    mavlink.MAV_TYPE_ANTENNA_TRACKER: mode_mapping_tracker,
+    # sub
+    mavlink.MAV_TYPE_SUBMARINE: mode_mapping_sub,
+}
+
+
+try:
+    # Allow for using custom mode maps by importing a JSON dict from
+    # "~/.pymavlink/custom_mode_map.json" and using it to extend the hard-coded
+    # AP_MAV_TYPE_MODE_MAP_DEFAULT dict.
+    from os.path import expanduser
+
+    _custom_mode_map_path = os.path.join("~", ".pymavlink", "custom_mode_map.json")
+    _custom_mode_map_path = expanduser(_custom_mode_map_path)
+    try:
+        with open(_custom_mode_map_path) as f:
+            _json_mode_map = json.load(f)
+    except:
+        print("Error: pymavlink custom mode file ('" + _custom_mode_map_path + "') is not valid JSON.")
```

Comment:
need to not throw an error if the file doesn't exist

Reply:
Now only catches `JSONDecodeError`. Any other failure to open or load the json reverts to default mode map.

Reply:
Error only thrown on failure to load json.